Add support for JOIN USING statements (#9629)
crate/crate: docs/sql/statements/select.rst

A ``joined_relation`` is a relation which joins two relations together. See
:ref:`sql_dql_joins` ::

>     relation { , | join_type JOIN } relation {[ ON join_condition ] | [USING (col_names)]}

Reviewer: `ON` and `USING` should be optional, so not sure if it is correct to put them into curly braces. Maybe something like this would work?
```suggestion
    relation { , | join_type JOIN } relation [ ON join_condition | USING (column_ident [, ...]) ]
```

>     relation { , | join_type JOIN } relation {[ ON join_condition ] | [USING (col_names)]}

Reviewer 2: I think it is:
```suggestion
    relation { , | join_type JOIN } relation [ { ON join_condition  |  USING (col_names) } ]
```

`[ .. ]`for the optional indication, `{ a | b | c }` to denote different options.

  The join_condition is not applicable for joins of type CROSS and must
  have a returning value of type ``boolean``.

:col_names:

>   A comma separated list of column names. The joining tables are both expected

Reviewer 2: ```suggestion
  A comma separated list of column names. The relations joined need to contain the columns specified.
```
